src/components/frame/v5/pages/UserProfilePage/partials/UserProfilePageForm.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect } from 'react';
+import clsx from 'clsx';
+import { useFormContext } from 'react-hook-form';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+
+import Toast from '~shared/Extensions/Toast';
+import Input from '~v5/common/Fields/Input';
+import Textarea from '~v5/common/Fields/Textarea';
+import Button from '~v5/shared/Button';
+import AvatarUploader from '~v5/common/AvatarUploader';
+import Avatar from '~v5/shared/Avatar';
+import { formatText } from '~utils/intl';
+import { UseAvatarUploaderProps } from '~v5/common/AvatarUploader/hooks';
+import { useAppContext, useMobile } from '~hooks';
+import { UserFragment, useUpdateUserProfileMutation } from '~gql';
+
+import styles from '../UserProfilePage.module.css';
+import { MAX_BIO_CHARS, MAX_DISPLAYNAME_CHARS } from '../consts';
+
+import LeftColumn from './LeftColumn';
+
+const displayName = 'v5.pages.UserProfilePage';
+
+interface Props {
+  user: UserFragment | null | undefined;
+  avatarUrl: string | null | undefined;
+  canChangeUsername: boolean;
+  daysTillUsernameChange: number;
+}
+
+const UserProfilePageForm = ({
+  user,
+  avatarUrl,
+  canChangeUsername,
+  daysTillUsernameChange,
+}: Props) => {
+  const {
+    formState: { dirtyFields, errors },
+    register,
+    setValue,
+    getValues,
+    reset,
+  } = useFormContext();
+  const { updateUser } = useAppContext();
+  const [updateAvatar] = useUpdateUserProfileMutation();
+  const isMobile = useMobile();
+  const { displayName: displayNameDirty } = dirtyFields;
+  const showNameMessage = displayNameDirty && !errors.displayName?.message;
+
+  useEffect(() => {
+    reset({
+      hasDisplayNameChanged: false,
+      bio: user?.profile?.bio || '',
+      displayName: user?.profile?.displayName || '',
+      location: user?.profile?.location || '',
+      website: user?.profile?.website || '',
+    });
+  }, [user, reset]);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const axiosUpload: UseAvatarUploaderProps['updateFn'] = async (
+    _,
+    thumbnail,
+    setProgress,
+  ) => {
+    axios.post(
```

Comment:
I know you didn't put it in here, but why is axios being used here? Why not strait up apollo ?

Reply:
No idea tbh, I wondered the same when I saw this

Reply:
this was something that page pro added that was broken when I refactored the `AvatarUploader` to be independent from the the user profile avatar uploading. I assumed it was something that had been agreed upon with us that this would be how uploading of files would eventually work. Feel free to remove it if this was not the case

Reply:
I don't remember this being agreed with us. And frankly I don't really like it. We need to find out more about this.